Conduit/Conduit/Controllers/FollowController.cs

```diff
@@ -0,0 +1,69 @@
+﻿using AutoMapper;
+using Conduit.Db;
+using Conduit.Dto;
+using Conduit.Model;
+using Conduit.Service;
+using Conduit.Service.passwordHasher;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace Conduit.Controllers
+{
+    [Route("Api/[Controller]")]
+    [ApiController]
+    public class FollowController: Controller
+    {
+        public IConfiguration _configuration;
+        public IUserRepositry _userRepositry;
+        public IFollowRepositry _followRepositry;
+
+        public FollowController(IConfiguration configuration, IMapper mapper, IUserRepositry userRepositry, IFollowRepositry followRepositry = null)
+        {
+            _configuration = configuration;
+            _userRepositry = userRepositry;
+            _followRepositry = followRepositry;
+        }
+
+        [HttpPost("Follow/",Name ="AddFavourite")]
+        [Authorize]
+        public async Task<IActionResult> addAfreind(Guid FriendId)
+
+        {
+            var data= getTokenInformation();
+            var userID = await _userRepositry.GetUserID(data.Email);
+            var Follow =await _followRepositry.followAfriend(userID, FriendId);
+
+            if (Follow)
+            {
+                return Ok("Followed !");
+            }           
+            return BadRequest();
+        }
+
+        private AuthModel getTokenInformation()
+        {
```

Comment:
I see this method was repeated multiple times, you can extract it to a service and only above call the method from service, so will get only one method, tbh it will be cleaner, consider if you need to update this method so you should check each class to update it instead of update once.